packages/gatsby-theme-vtex/src/components/ProductPage/SEO/index.tsx

```diff
@@ -13,22 +13,19 @@ const SEO: FC<ProductPageProps> = (props) => {
   const { product }: any = useAsyncProduct({ slug })
   const structuredProduct = useStructuredProduct(product, currency)
 
-  // There is no product. There is no point on generating tags yet
-  if (!product) {
-    return <SiteMetadataSEO {...props} />
-  }
-
   return (
     <>
       <SiteMetadataSEO {...props} />
-      <Helmet
-        script={[
-          {
-            type: 'application/ld+json',
-            innerHTML: JSON.stringify(structuredProduct),
-          },
-        ]}
-      />
+      {structuredProduct != null ? (
+        <Helmet
+          script={[
+            {
+              type: 'application/ld+json',
+              innerHTML: JSON.stringify(structuredProduct),
+            },
+          ]}
+        />
+      ) : null}
```

Comment:
Is there some reason for not using:
```suggestion
      {structuredProduct != null && (
        <Helmet
          script={[
            {
              type: 'application/ld+json',
              innerHTML: JSON.stringify(structuredProduct),
            },
          ]}
        />
      )}
```
?

Reply:
`structuredProduct != null` is false for `undefined`. This means we may end up rendering `undefined`, which is, as far as I know, not allowed by React

Reply:
🤔 I've used it a few times... [here](https://pt-br.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) it says React ignores if it returns false and doesn't render anything

Reply:
Cool !! I didn't know about. I'm changing it now